### ISSUE TITLE
Has LESS compilation logic been changed in WE 2.6.12?

### ISSUE THREAD

**Issue by miksh7** [madskristensen/WebEssentials2013#1926]
Let's assume the following file structure:

Folder /CSS/LESS:
File base.less:
@import "./app/_variables";
@import "./app/_layout";

Folder /CSS/LESS/APP:
files _variables.less and _layout.less with css markup and no import directives. 

As _layout.less depends on data from _variables.less then it failed to compile in VS2013 which is correct. When any changes are saved in _layout.less then CSS file update was done in the previous WE version. After upgrade to VS2015 Update 5 & WE 2.6.12 the css file is generated on save only if a less file could be compiled at the current hierarchy level, e.g. in this case when saving changes either in _layout.less or base.less.

Has LESS compilation logic been changed in WE 2.6.12?

**Comment by am11** [comment]
> Has LESS compilation logic been changed in WE 2.6.13?

No. There is another policy in play. If the target file exists, only then chained compilation causes the recompilation of the ancestry. To wit, you can save base.less once to make it emit base.css at desired location and then try changing the contents of _layout.less, it will then cause the dependent (base) to *re*compile.

This behavior is quite an old one. It is there to prevent generating undesired output files for the entire chain (except the partials) as it is pretty much hard to determine for WE to know which non-partial source files user intends to emit output for. If it generate output for all compilable files, then it will annoy people who like their source-control history tidy. So it was decided the best balance is to recompile dependency chain iff those have existing generated targets (at the desired location, in case CustomOutputDirectory was set).

**Comment by miksh7** [comment]
Thanks for details. Anyway, now the less file with "Compilation Error occurred" has NOT triggered generating of the output file.

**Comment by am11** [comment]
> has NOT triggered generating of the output file

It should if there exists an output file for base.less

**Comment by miksh7** [comment]
It should but not. See http://screencast.com/t/KYy1dAEwr25
This example is easy to reproduce. If needed I can send you the sample project.

**Comment by am11** [comment]
I have just upgraded to Windows 10 and VS2015 (with [WE2015](https://github.com/madskristensen/WebEssentials2015) and [NpmTaskRunner](https://github.com/madskristensen/NpmTaskRunner) <- configuring WE2013 node.js-based services to work with it, will share them once I get them working)  :)
Nonetheless, I have seen the problem. Thanks for the screencast!  will try to go back in time and debug with WE2013 SDK (in VM probably) and figure it out.

**Comment by thedavidbliss** [comment]
I can't seem to get my LESS to compile now with the latest update.

**Comment by josemahj** [comment]
the same problem here.....

main.less :
@import "test/_variables.less";
@import "test/_style.less";

_variables.less : 
@red: red;

_style.less: 
body {
    color: @red;
}

saving _style.less get:
/*

Compilation Error occurred (see error list to navigate to the error location):
Error found at line 2, column 11:


*/

**Comment by thedavidbliss** [comment]
I just updated mine with the latest release and it works great! No compilation errors at all. Release 2.6.36

**Comment by miksh7** [comment]
In my case, the issue is still there in 2.6.36
